### PR TITLE
A few cleanups of the provider build.infos

### PR DESCRIPTION
**Pull request by t8m**
Remove a TODO that is no longer relevant and
drop some more non-fips sources from the fips checksums.

### CODE REVIEW

**Comment by t8m** [comment]
The build failures are caused by libcommon sources being built after libdefault and the `prov/der....h` files are not yet generated. @levitte Is there any way how to easily add the dependency?

File: providers/common/der/build.info [hunk 1]
```diff
@@ -97,9 +97,6 @@ ENDIF
 
 #----- Conclusion
 
-# TODO(3.0) $COMMON should go to libcommon.a, but this currently leads
```
Comment (slontis) [inline providers/common/der/build.info:100]: Looks like this TODO was here for a reason :)

Comment (t8m) [reply providers/common/der/build.info:100]: Except it is not a problem of linking conflicts.

**Review by paulidale — APPROVED** [review]

**Review by slontis — APPROVED** [review]

**Comment by t8m** [comment]
This was approved 24h ago already. Merging.

**Comment by t8m** [comment]
Merged to master. Thank you for the reviews.

File: providers/common/der/build.info (hunk 1 above)
Comment (levitte) [reply providers/common/der/build.info:100]: So what is the problem according to you?

Comment (slontis) [reply providers/common/der/build.info:100]: keep reading..

File: providers/implementations/signature/build.info [hunk 2]
```diff
@@ -21,10 +21,10 @@ ENDIF
...
-DEPEND[rsa.o]=../../common/include/prov/der_rsa.h
-DEPEND[dsa.o]=../../common/include/prov/der_dsa.h
-DEPEND[ecdsa.o]=../../common/include/prov/der_ec.h
-DEPEND[eddsa.o]=../../common/include/prov/der_ecx.h
-DEPEND[sm2sig.o]=../../common/include/prov/der_sm2.h
+DEPEND[rsa_sig.o]=../../common/include/prov/der_rsa.h
+DEPEND[dsa_sig.o]=../../common/include/prov/der_dsa.h
+DEPEND[ecdsa_sig.o]=../../common/include/prov/der_ec.h
+DEPEND[eddsa_sig.o]=../../common/include/prov/der_ecx.h
+DEPEND[sm2_sig.o]=../../common/include/prov/der_sm2.h
```
Comment (levitte) [inline providers/implementations/signature/build.info:28]: :+1: :+1: :+1:

Comment (slontis) [reply providers/implementations/signature/build.info:28]: yep - you renamed files :)

File: providers/common/der/build.info (hunk 1 above)
Comment (levitte) [reply providers/common/der/build.info:100]: I would say, though, that *none* of the stuff here should have to go to libfips.a or libdefault.a.  It should really all go to libcommon.a...

...  and now that I look more closely, I see no reason why it's not.  I'll go digging a bit.

Comment (levitte) [reply providers/common/der/build.info:100]: or well, I can see reason why some of it would only end up in libdefault.a

Comment (levitte) [reply providers/common/der/build.info:100]: See #15338.  With *that*, this TODO is fixed in my view

File: providers/implementations/signature/build.info (hunk 2 above)
Comment (levitte) [reply providers/implementations/signature/build.info:28]: Picture me feeling massively guilty...

**Pull request closed without merging**